helix-term/src/commands/typed.rs

```diff
@@ -2008,6 +2008,10 @@ fn language(
 
     let id = doc.id();
     cx.editor.refresh_language_servers(id);
+    let doc = doc_mut!(cx.editor);
+    let diagnostics =
+        Editor::doc_diagnostics(&cx.editor.language_servers, &cx.editor.diagnostics, doc);
+    doc.replace_diagnostics(diagnostics, &[], None);
```

Comment:
I am afraid this is not sufficient since any document in a split (or in general any buffer open in the background) also need to be updated. It probably makes more sense to create an editor function that handles this for all documents (similar to `refresh_language_server`).

Reply:
Oh, ok, I see, so there can be multiple documents with different ids but the same path? Should such function have the signature `Editor::reinitialize_diagnostics(&mut self, document_path: &Path)` or something like that?

Isn't the rest of the code above this incorrect as well (e.g. `doc.set_language()`)?

Reply:
ah sorry I was tired and missrethis is for language and not general config reload in that case only refreshing a single document is fine. never mind

Reply:
Ok, that doesn't seem to be the case. So maybe I don't get what you mean. Do you mean that diagnostics need to be resent from the language server?

Reply:
Ah ok, had this still open, thanks for clearing it up :)